pandas/tests/frame/methods/test_update.py

```diff
@@ -146,3 +148,43 @@ def test_update_with_different_dtype(self):
 
         expected = DataFrame({"a": [1, 3], "b": [np.nan, 2], "c": ["foo", np.nan]})
         tm.assert_frame_equal(df, expected)
+
+    @given(dtype=np_st.integer_dtypes(endianness="="))
```

Comment:
Why is this needed?

Reply:
If you talk about endianness: Locally I there was an
> ValueError: Big-endian buffer not supported on little-endian compiler

so I used only native endian.

```
Falsifying example: test_update_with_subset_and_same_integer_dtype(
    self=<pandas.tests.frame.methods.test_update.TestDataFrameUpdate at 0x7f12171ac940>,
    dtype=dtype('>i2'),
)
Traceback (most recent call last):
  File "/home/jhuennemeyer/projects/pandas/pandas/tests/frame/methods/test_update.py", line 157, in test_update_with_subset_and_same_integer_dtype
    df.update(update)
  File "/home/jhuennemeyer/projects/pandas/pandas/core/frame.py", line 6947, in update
    other = other.reindex_like(self)
  File "/home/jhuennemeyer/projects/pandas/pandas/core/generic.py", line 4149, in reindex_like
    return self.reindex(**d)
  File "/home/jhuennemeyer/projects/pandas/pandas/util/_decorators.py", line 323, in wrapper
    return func(*args, **kwargs)
  File "/home/jhuennemeyer/projects/pandas/pandas/core/frame.py", line 4499, in reindex
    return super().reindex(**kwargs)
  File "/home/jhuennemeyer/projects/pandas/pandas/core/generic.py", line 4839, in reindex
    return self._reindex_multi(axes, copy, fill_value)
  File "/home/jhuennemeyer/projects/pandas/pandas/core/frame.py", line 4405, in _reindex_multi
    return self._reindex_with_indexers(
  File "/home/jhuennemeyer/projects/pandas/pandas/core/generic.py", line 4908, in _reindex_with_indexers
    new_data = new_data.reindex_indexer(
  File "/home/jhuennemeyer/projects/pandas/pandas/core/internals/managers.py", line 1305, in reindex_indexer
    new_blocks = [
  File "/home/jhuennemeyer/projects/pandas/pandas/core/internals/managers.py", line 1306, in <listcomp>
    blk.take_nd(
  File "/home/jhuennemeyer/projects/pandas/pandas/core/internals/blocks.py", line 1300, in take_nd
    new_values = algos.take_nd(
  File "/home/jhuennemeyer/projects/pandas/pandas/core/array_algos/take.py", line 72, in take_nd
    return _take_nd_ndarray(arr, indexer, axis, out, fill_value, allow_fill)
  File "/home/jhuennemeyer/projects/pandas/pandas/core/array_algos/take.py", line 116, in _take_nd_ndarray
    func(arr, indexer, out, fill_value)
  File "pandas/_libs/algos_take_helper.pxi", line 1181, in pandas._libs.algos.take_2d_axis0_int16_float64
ValueError: Big-endian buffer not supported on little-endian compiler
```


If you talk about the complete line: Test parameterization. Like I said, I would like to do that differently, but I'm limited but by hypothesis.

Reply:
For CI purposes, can you push a commit that removes the `given` decorator.

I know it will cause problems locally, but it would be good see what the CI says.

Reply:
Let me get this straight.
When you say:
> For CI purposes, can you push a commit that removes the `given` decorator.

I understand:
>Remove the complete line.

This line has its' purpose.
```
@given(dtype=np_st.integer_dtypes(endianness="="))
```
is the same as
```
@pytest.mark.parametrize("dtype", [np.dtype("=i1"), np.dtype("=i2"), np.dtype("=i4"), np.dtype("=i8")])
```
only that I don't have to worry about the exact parameters. That's what hypothesis does.
So removing it, just forces me to choose a specific dtype in the test. Probably I would simply use `int` and the error above is gone.
If you are still unclear about test parametrization or the hypothesis package let me know.
You can also check out [hypothesis](https://hypothesis.readthedocs.io/en/latest/quickstart.html) or [pytests docs  on that topic](https://docs.pytest.org/en/stable/fixture.html#fixture-parametrize).

By using hypothesis, I might have uncovered a very different bug.
The error, I got  locally, only arises if I choose any of these dtypes `[np.dtype(">i2"), np.dtype(">i4"), np.dtype(">i8")]` as
```
@given(dtype=np_st.integer_dtypes())
```
translates to
```
@pytest.mark.parametrize("dtype", [np.dtype("=i1"), np.dtype("=i2"), np.dtype("=i4"), np.dtype("=i8"), np.dtype(">i1"), np.dtype(">i2"), np.dtype(">i4"), np.dtype(">i8"), np.dtype("<i1"), np.dtype("<i2"), np.dtype("<i4"), np.dtype("<i8")])
```

So I will check out what your CI pipeline thinks about that.

Reply:
Resolved with [this comment](https://github.com/pandas-dev/pandas/pull/40219#discussion_r587485685).